core/graph/src/main/java/uk/gov/gchq/gaffer/graph/hook/NamedOperationResolver.java

```diff
@@ -37,79 +35,54 @@
-    private final int timeout;
-    private final TimeUnit timeUnit;
+    /**
+     * Default depth the resolver will go when checking for nested named operations
+     */
+    public static final int DEPTH_LIMIT_DEFAULT = 3;
 
-    public NamedOperationResolver(final String suffixNamedOperationCacheName) {
-        this(suffixNamedOperationCacheName, TIMEOUT_DEFAULT, TIME_UNIT_DEFAULT);
-    }
+    private final int depthLimit;
+    private final NamedOperationCache cache;
 
     @JsonCreator
     public NamedOperationResolver(
             @JsonProperty("suffixNamedOperationCacheName") final String suffixNamedOperationCacheName,
-            @JsonProperty("timeout") final int timeout,
-            @JsonProperty("timeUnit") final TimeUnit timeUnit) {
-        this(new NamedOperationCache(suffixNamedOperationCacheName), timeout, timeUnit);
+            @JsonProperty("depthLimit") final int depthLimit) {
+        this(new NamedOperationCache(suffixNamedOperationCacheName), depthLimit);
+    }
 
+    public NamedOperationResolver(final String suffixNamedOperationCacheName) {
+        this(suffixNamedOperationCacheName, DEPTH_LIMIT_DEFAULT);
     }
 
     public NamedOperationResolver(final NamedOperationCache cache) {
-        this(cache, TIMEOUT_DEFAULT, TIME_UNIT_DEFAULT);
+        this(cache, DEPTH_LIMIT_DEFAULT);
     }
 
-    public NamedOperationResolver(
-            final NamedOperationCache cache,
-            final int timeout,
-            final TimeUnit timeUnit) {
+    public NamedOperationResolver(final NamedOperationCache cache, final int depthLimit) {
         this.cache = cache;
-        this.timeout = timeout;
-        this.timeUnit = timeUnit;
+        this.depthLimit = depthLimit;
     }
 
     @JsonGetter("suffixNamedOperationCacheName")
     public String getSuffixCacheName() {
         return cache.getSuffixCacheName();
     }
 
-    @JsonGetter("timeout")
-    public int getTimeout() {
-        return timeout;
-    }
-
-    @JsonGetter("timeUnit")
-    public TimeUnit getTimeUnit() {
-        return timeUnit;
+    @JsonGetter("depthLimit")
+    public int getDepthLimit() {
+        return depthLimit;
     }
 
     @Override
     public void preExecute(final OperationChain<?> opChain, final Context context) {
-        try {
-            LOGGER.info("Resolving Named Operations with timeout: " + timeout);
-            CompletableFuture.runAsync(() -> resolveNamedOperations(opChain, context.getUser())).get(timeout, timeUnit);
-            LOGGER.info("Finished Named Operation resolver");
-        } catch (final ExecutionException | TimeoutException e) {
-            throw new GafferRuntimeException("ResolverTask did not complete: " + e.getMessage(), e);
-        } catch (final InterruptedException e) {
-            LOGGER.warn("Thread interrupted", e);
-            Thread.currentThread().interrupt();
-        }
-
+        opChain.updateOperations(resolveNamedOperations(opChain, context.getUser(), 0));
```

Comment:
depth hard coded to zero ?
I this correct, If this is correct can you stick a comment to explain why its zero.

Reply:
I believe this might show that there are no tests to check/prove that the depth limited is working or changed with fail and pass depths.

Reply:
This is correct if you follow the logic, the depth needs to be initialised somewhere this is the outer most and first call to the method so depth is currently zero. There is unit test to ensure the limit is working see line 140 in the `NamedOperationResolverTest`

Reply:
Foolish me, I miss read "depth" and "depthLimit"